### PR TITLE
[IndexTable] Use consistent padding around checkboxes

### PR DESCRIPTION
**Pull request by jesstelford**
Final rendered output should be identical.

Discovered when investigating https://github.com/Shopify/polaris/issues/11895

### CODE REVIEW

File: polaris-react/src/components/IndexTable/IndexTable.module.css [hunk 1]
```diff
@@ -1026,7 +1022,10 @@
     padding: var(--p-space-150);
 
     &:first-child {
-      padding-left: var(--p-space-300);
+      /* stylelint-disable-next-line -- specificity overrides */
+      padding-left: var(--pc-index-table-checkbox-offset-left);
+      /* stylelint-disable-next-line -- specificity overrides */
+      padding-right: var(--pc-index-table-checkbox-offset-right);
```
Comment (jesstelford) [inline polaris-react/src/components/IndexTable/IndexTable.module.css:1028]: This is `--p-space-200` (ie; `8px`). Before it was getting `--p-space-150` (`6px`) from above.

The 2px difference was being added to the inner width of the element, making it `20px` wide even though the checkbox is only `18px` wide.

Later, when the checkbox is rendered, it was then being wrapped in a `flex` container whos only job was to re-center the checkbox due to the added `2px`.

So, by setting the correct padding here, we can remove the unnecessary div later 👍

File: polaris-react/src/components/IndexTable/IndexTable.module.css [hunk 2]
```diff
@@ -1026,7 +1026,13 @@
     padding: var(--p-space-150);
 
     &:first-child {
-      padding-left: var(--p-space-300);
+      /* stylelint-disable-next-line -- specificity overrides */
+      padding-left: var(--pc-index-table-checkbox-offset-left);
+
+      .Table:not(.Table-unselectable) & {
+        /* stylelint-disable-next-line -- specificity overrides */
+        padding-right: var(--pc-index-table-checkbox-offset-right);
```
Comment (jesstelford) [inline polaris-react/src/components/IndexTable/IndexTable.module.css:1034]: This is `--p-space-200` (ie; `8px`). Before it was getting `--p-space-150` (`6px`) from above.

The 2px difference was being added to the inner width of the element, making it `20px` wide even though the checkbox is only `18px` wide.

Later, when the checkbox is rendered, it was then being wrapped in a `flex` container whos only job was to re-center the checkbox due to the added `2px`.

So, by setting the correct padding here, we can remove the unnecessary div later 👍

**Review by alex-page — APPROVED** [review]
Thanks Jess. Pinging @chloerice and @mrcthms for a second pair of eyes.

File: .changeset/seven-suns-love.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': patch
+---
+
+[IndexTable] Use consistent padding around checkboxes
```
Comment (chloerice) [inline .changeset/seven-suns-love.md]:
```suggestion
Removed extra padding around `IndexTable.Row` `Checkbox`
```

**Review by chloerice — APPROVED** [review]

**Review by mrcthms — APPROVED** [review]

**Pull request merged**